nncf/common/tensor_statistics/collectors.py

```diff
@@ -317,8 +318,10 @@ class MedianMADStatisticCollector(OfflineTensorStatisticCollector):
     Collector estimates median and median absolute deviation (MAD).
     """
 
-    def _prepare_statistics(self):
-        per_channel_history = get_per_channel_history(self._samples, list(self._reduction_shape), discard_zeros=True)
+    def _prepare_statistics(self) -> Tuple[npt.NDArray[Any], npt.NDArray[Any]]:
+        per_channel_history = get_per_channel_history(
+            self._samples, list(cast(Iterable[int], self._reduction_shape)), discard_zeros=True
```

Comment:
```suggestion
            self._samples, cast(List[int], list(self._reduction_shape)), discard_zeros=True
```

Reply:
`per_channel_history = get_per_channel_history(self._samples, cast(List[int], self._reduction_shape), discard_zeros=True)` this resolves the error